staging/src/k8s.io/apimachinery/pkg/api/meta/help.go

```diff
@@ -168,6 +191,19 @@ func EachListItem(obj runtime.Object, fn func(runtime.Object) error) error {
 // ExtractList returns obj's Items element as an array of runtime.Objects.
 // Returns an error if obj is not a List type (does not have an Items member).
 func ExtractList(obj runtime.Object) ([]runtime.Object, error) {
+	return extractList(obj, false)
+}
+
+// ExtractListWithAlloc returns obj's Items element as an array of runtime.Objects.
+// the difference from ExtractList is that an object of the same type is regenerated and assigned a value,
+// so that the returned Object list and obj have independent life cycles.
```

Comment:
This comment is a little misleading, since we only shallow-copy, and only when the items in the list are structs. The objects in the returned list are still somewhat related to the items in the list object.

Maybe a clearer phrasing would be:
```
This differs from ExtractList by returning shallow-copies of struct items to avoid retaining references to memory belonging to the list.
```

Reply:
Thanks for the suggestion, it has been fixed